dolphin/task_manager/tasks/task.py

```diff
@@ -30,23 +31,49 @@
 def set_synced_after(resource_type):
 
     @decorator.decorator
-    def _set_synced_after(f, *a, **k):
-        call_args = inspect.getcallargs(f, *a, **k)
+    def _set_synced_after(func, *args, **kwargs):
+        call_args = inspect.getcallargs(func, *args, **kwargs)
         self = call_args['self']
-        ret = f(*a, **k)
+        ret = func(*args, **kwargs)
         lock = coordination.Lock(self.storage_id)
         with lock:
-            storage = db.storage_get(self.context, self.storage_id)
-            storage[constants.DB.DEVICE_SYNC_STATUS] = utils.set_bit(
-                storage[constants.DB.DEVICE_SYNC_STATUS],
-                resource_type,
-                constants.SyncStatus.SYNCED)
-            db.storage_update(self.context, self.storage_id, storage)
+            try:
+                storage = db.storage_get(self.context, self.storage_id)
+            except exception.StorageNotFound:
+                LOG.warn('Storage %s not found when set synced'
+                         % self.storage_id)
+            else:
+                storage[constants.DB.DEVICE_SYNC_STATUS] = utils.set_bit(
+                    storage[constants.DB.DEVICE_SYNC_STATUS],
+                    resource_type,
+                    constants.SyncStatus.SYNCED)
+                db.storage_update(self.context, self.storage_id, storage)
         return ret
 
     return _set_synced_after
 
 
+def check_deleted(resource_type):
+
+    @decorator.decorator
+    def _check_deleted(func, *args, **kwargs):
+        call_args = inspect.getcallargs(func, *args, **kwargs)
+        self = call_args['self']
+        ret = func(*args, **kwargs)
+        self.context.read_deleted = 'yes'
+        try:
+            storage = db.storage_get(self.context, self.storage_id)
+        except exception.StorageNotFound:
+            LOG.warn('Storage %s not found when checking deleted'
+                     % self.storage_id)
+        else:
+            if storage['deleted'] != 0:
```

Comment:
Plz use constant instead of 0 here.

Reply:
Checked the code again, there is no need to have this 'if' sentence.
I have deleted it.